### PR TITLE
Add `VoxelPalette` iterator

### PR DESCRIPTION
**Pull request by roby2014**
# Description

Add VoxelPalette iterator

## Checklist

- [X] Self-review changes.
- [X] Evaluate impact on the documentation.
- [ ] Ensure test coverage.
- [ ] Write new samples.

### CODE REVIEW

**Comment by roby2014** [comment]
It's a fairly simple implementation, however, I'm a bit insecure on how should i test this? Engine tests?

**Review by RiscadoA — CHANGES_REQUESTED** [review]
Ty for working on this!

File: engine/include/cubos/engine/voxels/palette.hpp [hunk 1]
```diff
@@ -93,6 +93,28 @@ namespace cubos::engine
         /// @param similarity Minimum similarity for two materials to be merged.
         void merge(const VoxelPalette& palette, float similarity = 1.0F);
 
+        /// @brief Used to iterate over materials on the palette.
+        class Iterator
```
[the remark this answers is not in the document]
Comment (roby2014) [reply engine/include/cubos/engine/voxels/palette.hpp:97]: So, the definition will stay in the header file?

File: engine/include/cubos/engine/voxels/palette.hpp [hunk 2]
```diff
@@ -93,6 +93,28 @@ namespace cubos::engine
...
+        {
+        public:
+            Iterator(std::vector<VoxelMaterial>::iterator it);
+            Iterator& operator++();
+            VoxelMaterial& operator*();
+            bool operator==(const Iterator& other) const;
+            bool operator!=(const Iterator& other) const;
```
[the remark this answers is not in the document]
Comment (roby2014) [reply engine/include/cubos/engine/voxels/palette.hpp:104]: Yeah I forgot

**Comment by RiscadoA** [comment]
> It's a fairly simple implementation, however, I'm a bit insecure on how should i test this? Engine tests?

The tests are currently broken :( #564 Ig you can skip it for now

**Review by RiscadoA — CHANGES_REQUESTED** [review]

**Review by RiscadoA — APPROVED** [review]

**Pull request merged**